### PR TITLE
Disperse Storage Support

### LINKED ISSUE #524: Disperse Volume Support
* New Kadalu Storage Type(CRD) for Disperse Volume
* Disperse Type support in kubectl-kadalu
* Add Volfile Template for Disperse Volume Type
* New Storage Class(Default) for Disperse Volume
* Test Heal info command integration with Disperse Volume

### PR DESCRIPTION
**Pull request by aravindavk**
Added support for GlusterFS Disperse Volumes. Specify `--type=Disperse`
via CLI or `type: Disperse` in YAML.

New CLI Options:

```
--data NUMBER - Number of Disperse data Storage units
--redundancy NUMBER - Number of Disperse redundancy Storage units
```

YAML options:

```
...
  spec:
    disperse:
      data: 2
      redundancy: 1
...
```

Refer GlusterFS documentation for more details about Disperse Volumes.

https://docs.gluster.org/en/latest/Administrator-Guide/Setting-Up-Volumes/#creating-dispersed-volumes

Fixes: #524
Signed-off-by: Aravinda Vishwanathapura <aravinda@kadalu.io>

### CODE REVIEW

File: templates/operator.yaml.j2
Comment (amarts) [inline templates/operator.yaml.j2:89]: should we change the version of CRD for this?

Comment (aravindavk) [reply templates/operator.yaml.j2:89]: We can change. But not breaking the previously supported ones

File: helm/kadalu/templates/resources.yaml
Comment (leelavg) [inline helm/kadalu/templates/resources.yaml:26]: Not sure why CI failed to find the difference between manifests and helm templates, maybe CRD isn't being generated as apart of template in CI. Nevertheless, can you please also run `make gen-manifest`?

**Review by leelavg — COMMENTED** [review]
Will the docs be updated in a separate PR?

File: templates/operator.yaml.j2
Comment (amarts) [reply templates/operator.yaml.j2:89]: ok!

**Comment by amarts** [comment]
Can we plan to merge this after 0.8.2? I think its the time we should do 0.8.2 soon... has many fixes.

File: helm/kadalu/templates/resources.yaml
Comment (aravindavk) [reply helm/kadalu/templates/resources.yaml:26]: Done.

**Review by amarts — APPROVED** [review]

**Comment by amarts** [comment]
once the pylint errors are resolved, lets merge this.

**Comment by leelavg** [comment]
Please rebase, I already fixed pylint errors.

**Review by leelavg — APPROVED** [review]

**Pull request merged**
